concepts/0346-didcomm-between-two-mobile-agents/README.md

```diff
@@ -0,0 +1,123 @@
+# 0346: DIDCOMM BETWEEN TWO MOBILE AGENTS USING CLOUD AGENT MEDIATOR
+- Author: [Sukalpo Mitra](sukalpomitra@gmail.com)
+- Start Date: 2019-06-23
+
+## Status
+- Status: [PROPOSED](/README.md#proposed)
+- Status Date: 2019-06-23
+- Status Note: ???
+
+## Summary
+
+Explains how one mobile edge agent can send messages to another mobile edge agent through cloud agents. The sender edge agent also determines the route of the message. The recipient, on the other hand, can consume messages at its own pace and time.
+
+## Motivation
+[motivation]: #motivation
+
+The DIDCOMM between two mobile edge agents should be easy and intuitive for a beginner to visualize and to implement.
+
+## Scenario
+
+Alice sends a connection request message to Bob and Bob sends back an acceptance response. For simplicity's sake, we will only consider the cloud agents in play while sending and receiving a message for Alice.
+
+## Cloud Agent Registration Process
+
+A registration process is necessary for an edge agent to discover cloud agents that it can use to send a message through them. Cloud agents in the simplest form are routers hosted as a web application that solves the problem of availability by providing a persistent IP address. The Web server has a wallet of it's own storing its private key as a provisioning record, along with any information needed to forward messages to other agents. Alice wants to accept a connection invitation from Bob. But before doing so Alice needs to register herself with one or more cloud agents. The more cloud agents she registers with the more cloud agents she can use in transporting her message to Bob. To register herself with a cloud agent she visits the website of a cloud agent and simply scans a QR code.
+
+The cloud agent registration invite looks like below
+
+```JSON
+{​
+    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/didexchange/1.0/cloudagentregistrationinvitation",​
```

Comment:
I think that if you propose adding a new message type to the DID Exchange message family, you might want to coordinate with the authors of that protocol. It would probably make the protocol v 1.1 instead of 1.0, also. And note that we are in the process of switching from did:sov:.... as the prefix for these message types to https://didcomm.org/ as the prefix.

Reply:
@dhh1128 can you let me know whom I should contact?